### ISSUE TITLE
Error saving `PNG` files due to unsupported characters

### ISSUE THREAD

**Issue by Nenuphar12** [Pythagora-io/gpt-pilot#41]
## Description

While building a Flutter app using gpt-pilot, an error occurs when GPT-Pilot attempts to save generated PNG files.

## Environment

- GPT-Pilot Version: commit `38d5627`
- Flutter Version: 3.13.2
- OS: Windows 11

## Steps to reproduce

1. Launch gpt-pilot.
2. Command him to create a **Flutter** application.
3. execute the first command: `flutter create <project_name>`.

## Expected behavior

The command executes successfully and the generated files are saved in the database.

## Actual behavior

An error occurs when trying to save a `PNG` file.

## Error logs

```shell
Saving file \test_flutter\android\app\src\main\res\mipmap-hdpi\ic_launcher.png/ic_launcher.png
--- Logging error ---
...
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 3246, in execute_sql
    cursor.execute(sql, params or ())
ValueError: A string literal cannot contain NUL (0x00) characters.
```

(Note: Truncated for brevity.)

<details><summary>Full error logs</summary>
<pre><code>
Saving file \test_flutter\android\app\src\main\res\mipmap-hdpi\ic_launcher.png/ic_launcher.png
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 7117, in get
    return clone.execute(database)[0]
           ~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 4481, in __getitem__
    return self.row_cache[item]
           ~~~~~~~~~~~~~~^^^^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6702, in get_or_create
    return query.get(), False
           ^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 7120, in get
    raise self.model.DoesNotExist('%s instance matching query does '
database.models.file_snapshot.FileSnapshotDoesNotExist: <Model: FileSnapshot> instance matching query does not exist:
SQL: SELECT "t1"."id", "t1"."created_at", "t1"."updated_at", "t1"."app_id", "t1"."development_step_id", "t1"."file_id", "t1"."content" FROM "file_snapshot" AS "t1" WHERE ((("t1"."app_id" = %s) AND ("t1"."development_step_id" = %s)) AND ("t1"."file_id" = %s)) LIMIT %s OFFSET %s
Params: ['72dda6189f5a4272992f7a9f465369f0', 8, 56, 1, 0]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python311\Lib\logging\__init__.py", line 1113, in emit
    stream.write(msg + self.terminator)
  File "C:\Python311\Lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'charmap' codec can't encode character '\u03ae' in position 956: character maps to <undefined>
Call stack:
  File "C:\Users\nenup\tools\gpt-pilot\pilot\main.py", line 35, in <module>
    project.start()
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\Project.py", line 97, in start
    self.developer.start_coding()
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\agents\Developer.py", line 34, in start_coding
    self.implement_task(i, dev_task)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\agents\Developer.py", line 60, in implement_task
    self.execute_task(convo_dev_task, task_steps, development_task=development_task, continue_development=True)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\agents\Developer.py", line 78, in execute_task
    run_command_until_success(data['command'], data['timeout'], convo, additional_message=additional_message)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\cli.py", line 266, in run_command_until_success
    response = convo.send_message('dev_ops/ran_command.prompt',
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\AgentConvo.py", line 70, in send_message
    development_step = save_development_step(self.agent.project, prompt_path, prompt_data, self.messages, response)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\database\database.py", line 222, in save_development_step
    project.save_files_snapshot(development_step.id)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\Project.py", line 214, in save_files_snapshot
    file_snapshot, created = FileSnapshot.get_or_create(
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6708, in get_or_create
    return cls.create(**kwargs), True
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6577, in create
    inst.save(force_insert=True)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6787, in save
    pk = self.insert(**field_dict).execute()
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 1966, in inner
    return method(self, database, *args, **kwargs)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2037, in execute
    return self._execute(database)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2842, in _execute
    return super(Insert, self)._execute(database)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2553, in _execute
    cursor = self.execute_returning(database)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2560, in execute_returning
    cursor = database.execute(self)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 3254, in execute
    return self.execute_sql(sql, params)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 3243, in execute_sql
    logger.debug((sql, params))
Message: ('INSERT INTO "file_snapshot" ("id", "created_at", "updated_at", "app_id", "development_step_id", "file_id", "content") VALUES (%s, %s, %s, %s, %s, %s, %s) RETURNING "file_snapshot"."id"', ['0c054be44f8948b49d0ab5dcec7add7a', datetime.datetime(2023, 9, 5, 11, 10, 50, 724756), datetime.datetime(2023, 9, 5, 11, 10, 50, 724756), '72dda6189f5a4272992f7a9f465369f0', 8, 56, 'PNG\n\x1a\n\x00\x00\x00\nIHDR\x00\x00\x00H\x00\x00\x00H\x08\x03\x00\x00\x00b3Cu\x00\x00\x00\x19tEXtSoftware\x00Adobe ImageReadyqe<\x00\x00\x00PLTE\x00\x00\x00\x01N\x01W)T)T\x01V\x01W)FTT\x01V\x01W\x18v)T=\x002Y\x008d\x01+K\x010S\x01>n\x01>o\x01G\x7f\x01I\x01L\x01N\x01N\x01N\x01Q\x01R\x01R\x01S\x01U\x01U\x01V\x01V\x01W\x01W\x02;g\x02@p\x02Cv\x02I\x03M\x03O\x03P\x16h\x17o\x19x\x1a~\x1b\x1b\x1c)DT\x1a=\x00\x00\x00\x13tRNS\x00\x10\x10\x10\x10PP`````\x19\x10\x00\x00\x00IDATX\x0e@\x10@QT`\uf28c\x1d+?f\x08\x0bK"ή\x0fƹ79BILC\x0e9C\x0e9T/p!~\t0Nsx\t\x04%\\\'Jn;8\x16\'Ep^\tJ\x1cG\n8~)8";LIK\x12w\x1cI0N!\x1c%Q>Ȑ.;\x1d2\x12vf\x04\x19\x0b\x10Ԇ\x04ò3lHH2N\x7f\x03\x08JVo\x06dj\x1cpRy5`()V\x04#K$=\x00$#\x04;n\x00\x00\x00\x00IENDB`'])
Arguments: ()
Traceback (most recent call last):
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 7117, in get
    return clone.execute(database)[0]
           ~~~~~~~~~~~~~~~~~~~~~~~^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 4481, in __getitem__
    return self.row_cache[item]
           ~~~~~~~~~~~~~~^^^^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6702, in get_or_create
    return query.get(), False
           ^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 7120, in get
    raise self.model.DoesNotExist('%s instance matching query does '
database.models.file_snapshot.FileSnapshotDoesNotExist: <Model: FileSnapshot> instance matching query does not exist:
SQL: SELECT "t1"."id", "t1"."created_at", "t1"."updated_at", "t1"."app_id", "t1"."development_step_id", "t1"."file_id", "t1"."content" FROM "file_snapshot" AS "t1" WHERE ((("t1"."app_id" = %s) AND ("t1"."development_step_id" = %s)) AND ("t1"."file_id" = %s)) LIMIT %s OFFSET %s
Params: ['72dda6189f5a4272992f7a9f465369f0', 8, 56, 1, 0]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\nenup\tools\gpt-pilot\pilot\main.py", line 35, in <module>
    project.start()
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\Project.py", line 97, in start
    self.developer.start_coding()
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\agents\Developer.py", line 34, in start_coding
    self.implement_task(i, dev_task)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\agents\Developer.py", line 60, in implement_task
    self.execute_task(convo_dev_task, task_steps, development_task=development_task, continue_development=True)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\agents\Developer.py", line 78, in execute_task
    run_command_until_success(data['command'], data['timeout'], convo, additional_message=additional_message)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\cli.py", line 266, in run_command_until_success
    response = convo.send_message('dev_ops/ran_command.prompt',
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\AgentConvo.py", line 70, in send_message
    development_step = save_development_step(self.agent.project, prompt_path, prompt_data, self.messages, response)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot\database\database.py", line 222, in save_development_step
    project.save_files_snapshot(development_step.id)
  File "C:\Users\nenup\tools\gpt-pilot\pilot\helpers\Project.py", line 214, in save_files_snapshot
    file_snapshot, created = FileSnapshot.get_or_create(
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6708, in get_or_create
    return cls.create(**kwargs), True
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6577, in create
    inst.save(force_insert=True)
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 6787, in save
    pk = self.insert(**field_dict).execute()
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 1966, in inner
    return method(self, database, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2037, in execute
    return self._execute(database)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2842, in _execute
    return super(Insert, self)._execute(database)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2553, in _execute
    cursor = self.execute_returning(database)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 2560, in execute_returning
    cursor = database.execute(self)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 3254, in execute
    return self.execute_sql(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nenup\tools\gpt-pilot\pilot-env\Lib\site-packages\peewee.py", line 3246, in execute_sql
    cursor.execute(sql, params or ())
ValueError: A string literal cannot contain NUL (0x00) characters.
</code></pre>
</details>

**Comment by LeonOstrez** [comment]
Thanks for opening the issue.

I added this bug to the list and I'll keep you updated here once we solve it.

**Comment by soccerwithag** [comment]
Same error, keeps trying to save file "Saving file /.DS_Store" ... "ValueError: A string literal cannot contain NUL (0x00) characters."

**Comment by nalbion** [comment]
@LeonOstrez instead of saving files to the DB, could we just `git commit` ( #199 )

This would also fix #139 , #169
